components/Typography/tests/unit/MaterialScalableFontTests.m

```diff
@@ -0,0 +1,266 @@
+    UIContentSizeCategoryExtraLarge : @0,
+    UIContentSizeCategoryExtraExtraLarge : @0,
+    UIContentSizeCategoryExtraExtraExtraLarge : @0,
+    UIContentSizeCategoryAccessibilityMedium : @-1,
+    UIContentSizeCategoryAccessibilityLarge : @-1,
+    UIContentSizeCategoryAccessibilityExtraLarge : @-1,
+    UIContentSizeCategoryAccessibilityExtraExtraLarge : @-1,
+    UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @-1,
+  };
+
+  font.mdc_scalingCurve = scalingCurve;
+
+  // Note that scaling curve is 0 @ UIContentSizeCategoryExtraExtraExtraLarge so this font should be
+  // the same size.
+  UIFont *zeroScaledFont =
+      [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryExtraExtraExtraLarge];
+
+  // Note that scaling curve is -1 @ UIContentSizeCategoryAccessibilityExtraExtraExtraLarge so this
+  // font should be the same size.
+  UIFont *negativeScaledFont =
+      [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+
+  // Then
+  XCTAssert([font mdc_isSimplyEqual:zeroScaledFont]);
+  XCTAssert([font mdc_isSimplyEqual:negativeScaledFont]);
+}
+
+- (void)testIncompleteScalingCurve {
+  // Given
+  UIFont *font = [UIFont systemFontOfSize:20.0];
+
+  const CGFloat curvePointSize = 12.0;
+
+  // This curve is missing all values over Medius
+  NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+    UIContentSizeCategoryExtraSmall : @(curvePointSize),
+    UIContentSizeCategorySmall : @(curvePointSize),
+    UIContentSizeCategoryMedium : @(curvePointSize),
+  };
+
+  font.mdc_scalingCurve = scalingCurve;
+
+  UIFont *mediumScaledFont = [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryMedium];
+
+  UIFont *missingCurveScaledFont =
+      [font mdc_scaledFontForSizeCategory:UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+
+  // Then
+  XCTAssertTrue(MDCCGFloatEqual(mediumScaledFont.pointSize, curvePointSize));
+  XCTAssert([font mdc_isSimplyEqual:missingCurveScaledFont]);
+}
+
+- (void)testValueScaling {
+  // Given
+  UIFont *originalFont = [UIFont systemFontOfSize:20.0];
+  CGFloat originalValue = 10.0;
+
+  MDCFontScaler *scaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleHeadline1];
+  UIFont *scalableFont = [scaler scalableFontWithFont:originalFont];
+
+  // When
+  UIFont *defaultFont = [scalableFont mdc_scaledFontAtDefaultSize];
+  UIFont *currentFont = [scalableFont mdc_scaledFontForCurrentSizeCategory];
+
+  CGFloat scalerScaledValue = [scaler scaledValueForValue:originalValue];
+  CGFloat fontScaledValue = [scalableFont mdc_scaledValueForValue:originalValue];
+
+  CGFloat sizeScaleFactor = currentFont.pointSize / defaultFont.pointSize;
+  CGFloat scalerScaleFactor = scalerScaledValue / originalValue;
+  CGFloat fontScaleFactor = fontScaledValue / originalValue;
+
+  //Then
+  XCTAssertTrue(MDCCGFloatEqual(sizeScaleFactor, scalerScaleFactor));
+  XCTAssertTrue(MDCCGFloatEqual(scalerScaleFactor, fontScaleFactor));
```

Comment:
Tip: Since you are relying on the internal determination of "current size category", consider using a Test Double and setting a specific value for that method to return so you can verify the behavior of the other methods.

Reply:
Changed per offline discussion.